### PR TITLE
Feature/ia ticket21 flights basic

### LINKED ISSUE #21: As a user, I want to be able to determine my carbon emissions from the flights I take
**__User Story Description__**
As a user
I want to be able to determine my carbon emissions from the flights I take
So I can inform the cost of my use

**__Steps to Follow (optional)__**
- [x] Research calculations needed for Carbon API
- [x] Communicate to frontend what questions need to be added to UI
- [x] Flight/FlightLeg Model
- [x] Flight/FlightLeg Schema
- [x] Add additional steps as necessary

**__Additional Considerations__**
https://docs.carboninterface.com/#/?id=flight - flight estimate request will accept passenger flight details and compute the carbon emissions for the trip.

https://www.goflightlabs.com/ - Airport Codes API

After determining what data is need for Carbon API request, communicate this with Issue #44

### LINKED ISSUE #44: As a user, I want to be able to determine my carbon emissions from the flights I take
**__User Story Description__**
As a user
I want to be able to determine my carbon emissions from the flights I take
So I can inform the cost of my use

**__Steps to Follow (optional)__**
- [x] Create form on '/travel' page in the driving tab
- [x] Communicate with Developer on Issue #21 for questions to put in UI
- [x] Form on submit should send correct data to backend
- [x] Add additional steps as necessary

**__Additional Considerations__**
Keep in mind the backend version of this ticket #21

### PR DESCRIPTION
**Pull request by ialej001**
Added complete end-to-end flight calculator:
- Added flight calculator to travel page inside a tabbed component.
- Added 4 child components that render each section of the flight calculator UI into the components folder under 'Flight'.
- Added as much client side validation to minimize the amount of potential CarbonAPI server side errors.
- Set up flight schemas and types
- Set up tRPC router endpoint for unregistered users and merged into the tRPC controller

Note: To whomever takes up the driving ticket: replace my flight component with your parent component in the driving tab in the travel page.

Closes #21, #44

### CODE REVIEW

File: src/pages/travel.tsx [hunk 1]
```diff
@@ -9,9 +14,45 @@ export default function TravelPage() {
...
+            <FlightQuestionaire />
+          </Tabs.Item>
+          <Tabs.Item title="Driving" icon={FaCarAlt}>
+            <p>Your Driving parent component here</p>
+            <FlightQuestionaire />
+          </Tabs.Item>
+        </Tabs.Group>
+      </div>
+
+      {/* <div className='flex mx-auto min-w-fit w-96 justify-center'>
```
Comment (TiciaD) [inline src/pages/travel.tsx]: We might want to remove commented out code just to make the final file look a little cleaner

File: src/pages/travel.tsx [hunk 2]
```diff
@@ -9,9 +14,45 @@ export default function TravelPage() {
         <meta name="viewport" content="initial-scale=1.0, width=device-width" />
       </Head>
       <nav className="flex justify-center gap-4">
-        <Link href='/'>Home</Link>
+        <Link href="/">Home</Link>
       </nav>
-      <h2>Travel Page</h2>
+
+      <div className='flex justify-center'>
+        <Image src={wateringPlantsImage} />
```
Comment (TiciaD) [inline src/pages/travel.tsx]: We should probably add an `alt` description for accessibility purposes, even though the image is just decorative

File: src/components/Flight/FlightItinerarySummary.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+interface FlightItinerarySummaryProps {
+  data: FlightLeg[];
+}
+const FlightItinerarySummary = ({ data }: FlightItinerarySummaryProps) => {
+  return (
+    <div>
+      {data.map((leg, index) => {
+        return (
+          <>
+            <p>
```
Comment (TiciaD) [inline src/components/Flight/FlightItinerarySummary.tsx]: You'll have to add a key in here somewhere so we don't keep getting this error in the console:
Screenshot 2022-08-21 141420

**Review by TiciaD — CHANGES_REQUESTED** [review]
Other than those few cleanup comments, it looks good Ivan! Thanks for putting this together so quickly so we can have a reference for other pages!

File: src/pages/travel.tsx (hunk 2 above)
Comment (ialej001) [reply src/pages/travel.tsx]: added alt message

File: src/pages/travel.tsx (hunk 1 above)
Comment (ialej001) [reply src/pages/travel.tsx]: removed comment block

File: src/components/Flight/FlightItinerarySummary.tsx (hunk 3 above)
Comment (ialej001) [reply src/components/Flight/FlightItinerarySummary.tsx]: added keys for the p tags

**Review by TiciaD — APPROVED** [review]
Thanks for addressing those cleanup comments! Looks great Ivan!

**Review by amandaalexandre — APPROVED** [review]

**Comment by amandaalexandre** [comment]
Thanks so much! I'll take a look at it!

**Pull request merged**